tools/operator/connectionprofile/connectionprofile.go

```diff
@@ -91,6 +100,221 @@ func (c ConnProfile) GenerateConnProfilePerOrg(orgName string) error {
+	for c := range caMAP {
+		ca := caMAP[c]
+		caliperCA := networkspec.CaliperCertificateAuthority{URL: ca.URL, CAName: ca.CAName, TLSCACerts: ca.TLSCACerts, HTTPOptions: ca.HTTPOptions}
+		caReg := networkspec.CARegistrar{EnrollID: ca.Registrar.EnrollID, EnrollSecret: ca.Registrar.EnrollSecret}
+		caliperCA.Registrar = append(caliperCA.Registrar, caReg)
+		caliperCAs[c] = caliperCA
+	}
+
+	return caliperCAs, nil
+}
+
+func (c ConnProfile) orgs2caliper(orgMAP map[string]networkspec.Organization) (map[string]networkspec.CaliperOrganization, error) {
+	caliperOrgs := make(map[string]networkspec.CaliperOrganization)
+	for o := range orgMAP {
+		org := orgMAP[o]
+		caliperOrg := networkspec.CaliperOrganization{MSPID: org.MSPID, Peers: org.Peers, CertificateAuthorities: org.CertificateAuthorities, AdminPrivateKey: org.AdminPrivateKey, SignedCert: org.SignedCert}
+		caliperOrgs[o] = caliperOrg
+	}
+
+	return caliperOrgs, nil
+}
+
+func (c ConnProfile) connprofile2caliper(connProfile ConnProfile) (CaliperConnProfile, error) {
+	caliperConnProfile := CaliperConnProfile{}
+
+	var err error
+	caliperConnProfile.Orderers, err = c.orderers2caliper(connProfile.Orderers)
+	if err != nil {
+		return caliperConnProfile, err
+	}
+
+	caliperConnProfile.Peers, err = c.peers2caliper(connProfile.Peers)
+	if err != nil {
+		return caliperConnProfile, err
+	}
+
+	caliperConnProfile.CA, err = c.ca2caliper(connProfile.CA)
+	if err != nil {
+		return caliperConnProfile, err
+	}
+
+	caliperConnProfile.Organizations, err = c.orgs2caliper(connProfile.Organizations)
+	if err != nil {
+		return caliperConnProfile, err
+	}
+
+	caliperConnProfile.Config = connProfile.Config
+
+	return caliperConnProfile, nil
+}
+
+func (c ConnProfile) clientCaliper(orgName string) (map[string]networkspec.CaliperClient, error) {
+	clients := make(map[string]networkspec.CaliperClient)
+	var client networkspec.CaliperClient
+
+	client.Client.Organization = orgName
+	client.Client.CredentialStore.Path = fmt.Sprintf("/tmp/%s", orgName)
+	client.Client.CredentialStore.CryptoStore.Path = fmt.Sprintf("/tmp/%s", orgName)
+	client.Client.Conenction.Timeout.Peer.Endorser = 300
+	client.Client.Conenction.Timeout.Peer.EventHub = 600
+	client.Client.Conenction.Timeout.Peer.EventReg = 300
+	client.Client.Conenction.Timeout.Orderer = 300
+	peerOrgsPath := paths.PeerOrgsDir(c.Config.ArtifactsLocation)
+
+	//privateKey
+	keystorePath := paths.JoinPath(peerOrgsPath, fmt.Sprintf("%s/users/User1@%s/msp/keystore", orgName, orgName))
+	privKeyFile, err := ioutil.ReadDir(keystorePath)
+	if err != nil {
+		return clients, err
+	}
+	privKeyPath := paths.JoinPath(keystorePath, fmt.Sprintf("%s", privKeyFile[0].Name()))
+	cert, err := c.GetCertificateFromFile(privKeyPath)
+	if err != nil {
+		return clients, err
+	}
+	client.Client.ClientPrivateKey.Pem = cert
+
+	//signedCerts
+	signCertsPath := paths.JoinPath(peerOrgsPath, fmt.Sprintf("%s/users/User1@%s/msp/signcerts", orgName, orgName))
```

Comment:
Instead of generating the path here and above, instead create the path to the `/msp` dir and then just build the keystore and the signcert dir when you do the `paths.JoinPath`

Reply:
ok